### PR TITLE
[Core] Use polled waiting on ChibiOS platforms that support it

### PR DESCRIPTION
**Pull request by KarlK90**
## Description

Due to context switching overhead waiting a very short amount of time on a sleeping thread is often not accurate and in fact not usable for timing critical usage i.e. in a driver. Thus we use polled waiting for ranges in the us range on platforms that support it instead. The fallback is the thread sleeping mechanism.

This includes:

* ARM platforms with CYCCNT register (ARMv7, ARMv8) this is incremented at CPU clock frequency
* GD32VF103 RISC-V port with CSR_MCYCLE register this is incremented at CPU clock frequency
* RP2040 ARMv6 port which uses the integrated timer peripheral which is incremented with a fixed 1MHz frequency

## Comparison busy waiting vs sleeping

Waiting values are 0, 1, 5, 10, 25, 50, 100, 150, 200, 500 and 1000 microseconds with 10us spacing. Pink is busy waiting, green is sleeping by yielding the thread.

### STM32F411

stm32f411

<span style="color:pink">busy waiting</span>

* minimal busy waiting: **1us**
* minimal sleeping time: **111us**

### RP2040

rp2040

* minimal busy waiting: **1us**
* minimal sleeping time: **28us**

### GD32VF103

gd32vf103

* minimal busy waiting: **1us**
* minimal sleeping time: **189us**

Matrix scanning frequency on DZZY keypad with 5x6 matrix:

* busy waiting: **4.6k** (575%!)
* thread sleeping: **800**

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Overhead in pointing device drivers and matrix scanning

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: platforms/chibios/_wait.h [hunk 1]
```diff
@@ -30,6 +30,11 @@
 
 #ifdef WAIT_US_TIMER
 void wait_us(uint16_t duration);
+#elif PORT_SUPPORTS_RT == TRUE
+#    define wait_us(us)                                            \
+        do {                                                       \
+            chSysPolledDelayX(US2RTC(REALTIME_COUNTER_CLOCK, us)); \
```
Comment (zvecr) [inline platforms/chibios/_wait.h:36]: Does `chSysPolledDelayX` interpret zero as infinite?

Comment (KarlK90) [reply platforms/chibios/_wait.h:36]: Nope it just skips the wait http://chibiforge.org/doc/21.11/full_rm/chsys_8c_source.html#l00492

File: platforms/chibios/drivers/serial.c [hunk 2]
```diff
@@ -18,9 +18,6 @@
 //   this method seems to be more accurate than GPT timers
 #if PORT_SUPPORTS_RT == FALSE
 #    error "chSysPolledDelayX method not supported on this platform"
-#else
-#    undef wait_us
-#    define wait_us(x) chSysPolledDelayX(US2RTC(CPU_CLOCK, x))
```
Comment (zvecr) [inline platforms/chibios/drivers/serial.c:23]: Can we put this back for now so that it doesn't accidentally use `WAIT_US_TIMER`

**Review by zvecr — COMMENTED** [review]
Otherwise LGTM

File: platforms/chibios/drivers/serial.c (hunk 2 above)
Comment (KarlK90) [reply platforms/chibios/drivers/serial.c:23]: Yes, will do that!

Comment (KarlK90) [reply platforms/chibios/drivers/serial.c]: Done.

File: platforms/chibios/chibios_config.h [hunk 3]
```diff
@@ -109,3 +112,12 @@
 #    define PAL_OUTPUT_TYPE_PUSHPULL PAL_HT32_MODE_DIR
 #    define PAL_OUTPUT_SPEED_HIGHEST 0
 #endif
+
+#if defined(MCU_MIMXRT1062)
+#    include "clock_config.h"
+#    define CPU_CLOCK BOARD_BOOTCLOCKRUN_CORE_CLOCK
+#endif
```
Comment (zvecr) [inline platforms/chibios/chibios_config.h:119]: A bit pedantic, but can we move this under the `MCU_KINETIS` block so the teensy stuff is next to each other

Comment (KarlK90) [reply platforms/chibios/chibios_config.h]: No problem, done.

**Review by zvecr — APPROVED** [review]

**Pull request merged**